### ISSUE TITLE
💡 [REQUEST] Add the latest PyTorch features into basics Autograd tutorial

### ISSUE THREAD

**Issue by svekars** [pytorch/tutorials#3119]
### 🚀 Describe the improvement or the new tutorial

The Autograd [tutorial](https://pytorch.org/tutorials/beginner/basics/autogradqs_tutorial) does a good job of covering the basic graph creation and then running .backward(). However, there are new features from the latest PyTorch that can be highlighted in the tutorial.

cc: @soulitzer 

### Existing tutorials on this topic

https://pytorch.org/tutorials/beginner/basics/autogradqs_tutorial

**Comment by soulitzer** [comment]
> The Autograd [tutorial](https://pytorch.org/tutorials/beginner/basics/autogradqs_tutorial) does a good job of covering the basic graph creation and then running .backward().

I agree the overall the tutorial is in a good place as the first things someone reads as they are starting out (e.g. how to do basic backprop on a module with .backward().  

A lot of new features also require some more concepts to be introduced, and we can either do that by repositioning this tutorial to also cover intermediate concepts OR having a separate  "intermediate autograd" tutorial.

If we continue to keep this as a "starting out" tutorial, we should be careful to have a balance between providing enough information for the user to read more, but also keep it approachable, e.g., briefly mention/link the newer features related to each section in the form of optional notes. 

I think some improvements for now we could make are:

- disabling gradient tracking should link to the autograd mechanics doc that goes into disabling gradient tracking more in depth

- for the computational graph section, we can introduce some more tools to help visualize that better:
[TORCH_LOGS="+autograd"](https://dev-discuss.pytorch.org/t/highlighting-a-few-recent-autograd-features-h2-2023/1787
), [torchviz](https://github.com/szagoruyko/pytorchviz)

**Comment by ParagEkbote** [comment]
Could I please take this issue as part of the current docathon?

cc: @svekars
